caravel/models.py

```diff
@@ -577,7 +577,19 @@ def query(  # sqla
                 select_exprs += [timestamp_grain]
                 groupby_exprs += [timestamp_grain]
 
-            tf = '%Y-%m-%d %H:%M:%S.%f'
+            # UGLY: I guess correct way is to delegate on SQLAlchemy dialect
+            # UPDATE: Datetime depends on each dialect and I haven't found an easy way to manage
+            #         Maybe we can allow user to define its custome format at Database definition
+            def get_dtformat(type):
```

Comment:
FYI, you should never overwrite system words like `type`, the convention is to suffix with underscore, or just using something more specific.